### PR TITLE
VIX-3113

### PR DESCRIPTION
**Pull request by johncbaur**

### CODE REVIEW

File: Modules/Effect/Bars/Bars.cs [hunk 1]
```diff
@@ -2573,8 +2572,11 @@ private void GetRotatedPosition(ref int x, ref int y, double angle, bool swapXY)
 	        // Transform (rotate) the point
 	        System.Windows.Point transformedPoint = rt.Transform(tempPoint);
 
-	        // Updated the x and y coordinates for the rotation
-	        x = (int)Math.Round(transformedPoint.X);
+            //transformedPoint.X -= xOffset;
+            //transformedPoint.Y -= yOffset;
+
```
Comment (jeffu231) [inline Modules/Effect/Bars/Bars.cs]: Looks like the previous commit removed these two commented lines, and then they are put back here. Was the intent to remove them?

File: Modules/Effect/Bars/Bars.cs [hunk 2]
```diff
@@ -2256,24 +2256,23 @@ private int CalculateZigZagYTilePosition(bool incrementing, int y, int movementY
             if (incrementing)
             {
                 // Calculate the Y position within the tile
-                yTile = (y + movementY) % heightOfTile;
+                yTile = (y + movementY) % heightOfTile;
```
Comment (jeffu231) [inline Modules/Effect/Bars/Bars.cs:2259]: Looks like this was just a spacing change at the end of the line. Was there some other intended change here?

**Review by jeffu231 — COMMENTED** [review]
Any concerns of impact to existing effects, or would this only impact ones that were not working like the reporter?

**Comment by johncbaur** [comment]
The only method I am intending to change is CalculateZigZagyTilePosition.
I had some commented out code in my local copy that snuck into my commit.
I tried to back that out on the second commit.  I don't see this breaking
any other usages of the effect.  I had guard logic when decrementing to
make sure the index did not go less than zero.  I did not have the guard
for the incrementing half of the if because I probably thought it was
impossible and might improve performance if it was omitted.  The reporter's
effect went down the incrementing path but the increment was negative.  I
believe this is due to the Speed selection (-15 to 15).  Sorry for the bugs
that have snuck into this effect, I did do a significant amount of testing
last Christmas break but it was using my profile.  Hopefully in the end the
added functionality was worth the risk of the introduction of bugs.

**Pull request merged**